csFastFloat/Utils/Utils.cs

```diff
@@ -22,8 +22,42 @@ public value128(ulong h, ulong l) : this()
 
   public static class Utils
   {
-    // Next function can be micro-optimized, but compilers are entirely
-    // able to optimize it well.
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static uint parse_eight_digits_unrolled(ulong val)
+    {
+      const ulong mask = 0x000000FF000000FF;
+      const ulong mul1 = 0x000F424000000064; // 100 + (1000000ULL << 32)
+      const ulong mul2 = 0x0000271000000001; // 1 + (10000ULL << 32)
+      val -= 0x3030303030303030;
+      val = (val * 10) + (val >> 8); // val = (val * 2561) >> 8;
+      val = (((val & mask) * mul1) + (((val >> 16) & mask) * mul2)) >> 32;
+      return (uint)val;
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    unsafe internal static uint parse_eight_digits_unrolled(byte* chars)
+    {
+      ulong val = Unsafe.ReadUnaligned<ulong>(chars);
```

Comment:
This could be done more C-like by pointer-casting too:
```suggestion
      ulong val = *(ulong*)chars;
```
It's a matter of taste here.

Reply:
Is the second version safe even when the pointer is not aligned?

Reply:
Yes -- on all implemented versions of .NET

In theory there could be a future runtime for .NET where it wouldn't be safe, but that would break a lot of code (even with .NET runtime itself), so that is very unlikely to happen.